yarn-project/types/src/l2_block.ts

```diff
@@ -659,6 +654,22 @@ export class L2Block {
     );
   }
 
+  /**
+   * A lightweight method to get the tx hash of a tx in the block.
+   * @param txIndex - the index of the tx in the block
+   * @returns a hash of the tx, which is the first nullifier in the tx
+   */
+  getTxHash(txIndex: number): TxHash {
+    this.assertIndexInRange(txIndex);
+
+    const firstNullifier = this.newNullifiers.slice(
+      txIndex * MAX_NEW_NULLIFIERS_PER_TX,
+      (txIndex + 1) * MAX_NEW_NULLIFIERS_PER_TX,
+    )[0];
```

Comment:
I think this is equivalent and more efficient since it won't create a new array 
```suggestion
    const firstTxNullifier = this.newNullifiers[txIndex * MAX_NEW_NULLIFIERS_PER_TX];
```